fleetspeak/src/server/components/proto/fleetspeak_components/config.proto

```diff
@@ -75,3 +78,13 @@ message AdminConfig {
   // "localhost:1234". Required.
   string listen_address = 1;
 }
+
+message StatsConfig {
+  // Stats collector to be used. Optional, but recommended.
+  // Defaults to no stats collector.
+  string collector = 1;
+
+  // The bind address to listen on for connections, e.g. ":443" or
+  // "localhost:1234". Optional, defaults to ":2112".
```

Comment:
I agree that mentioning port 443 might be misleading and/or confusing, so I'll make sure to stick to an obviously arbitrary port like 2112 and get rid of the notion of a "default" port to fit the logic.
The reason for using 2112 as default was merely inspired by the Prometheus docs to [instrumenting a Go application](https://prometheus.io/docs/guides/go-application/#how-go-exposition-works) (such as Fleetspeak), where the example port was 2112 as well.

Reply:
Mentioning it as a default probably makes sense. But the port shouldn't be publicallly accessible, so it is probably best if it requires some intent on the part of the admin setting things up to open it. Especially if it is bound to a public address ":2112" instead of "localhost:2112", we should make it hard to miss the need to firewall it.

Reply:
> Mentioning it as a default probably makes sense. But the port shouldn't be publicallly accessible, so it is probably best if it requires some intent on the part of the admin setting things up to open it. Especially if it is bound to a public address ":2112" instead of "localhost:2112", we should make it hard to miss the need to firewall it.

I'm not sure what you mean - do you think that in the comments, an example should be in the form of "localhost:2112" rather than plain ":2112"?

Reply:
http.ListenAndServer(":2112") listens on all interfaces. http.ListenAndServer("localhost:2112") listens on the loopback interface. The decision has security implications - the first is safer, and great for demos, but obviously isn't what you want in a large install.

In some installations the host might have both a public and private interface, with the private interface/subnet used to communicate purely with other servers. In this case, they might want to bind to "<private address>:2112".

This is why I try to always use bind address as the controlling parameter, not port number. This standard is suggested by the go libraries, and it is a good one.

So yes, the method you linked should change and the protocol buffer field should be renamed accordingly.